python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -167,130 +162,27 @@ def _getRoot(self, cond=None):
-            return
-
-        if _srcobj._schema is None:
-            self._schema = None
-            return
-
-        self._actually_copyFrom(_srcobj, _ignore_atts)
-
-        ## Fix some objects losing parent knowledge
-        src_dict = srcobj.__dict__
-        for key, val in src_dict.iteritems():
-            this_attr = getattr(srcobj, key)
-            if isinstance(this_attr, Node) and key not in do_not_copy:
-                #logger.debug("k: %s  Parent: %s" % (key, (srcobj)))
-                this_attr._setParent(srcobj)
-
-    def _actually_copyFrom(self, _srcobj, _ignore_atts):
-
-        for name, item in self._schema.allItems():
-            if name in _ignore_atts:
-                continue
-
-            #logger.debug("Copying: %s : %s" % (name, item))
-            if name == 'application' and hasattr(_srcobj.application, 'is_prepared'):
-                _app = _srcobj.application
-                if _app.is_prepared not in [None, True]:
-                    _app.incrementShareCounter(_app.is_prepared.name)
+    # mark object as "dirty" and inform the registry about it
+    # the registry is always associated with the root object
+    def _setDirty(self):
+        """ Set the dirty flag all the way up to the parent"""
+        self._dirty = True
+        parent = self._getParent()
+        if parent is not None:
+            parent._setDirty()
 
-            if not self._schema.hasAttribute(name):
-                #raise ValueError('copyFrom: incompatible schema: source=%s destination=%s'%(_getName(_srcobj), _getName(self)))
-                if not hasattr(self, name):
-                    setattr(self, name, self._schema.getDefaultValue(name))
-                this_attr = getattr(self, name)
-                if isinstance(this_attr, Node) and name not in do_not_copy:
-                    this_attr._setParent(self)
-            elif not item['copyable']: ## Default of '1' instead of True...
-                if not hasattr(self, name):
-                    setattr(self, name, self._schema.getDefaultValue(name))
-                this_attr = getattr(self, name)
-                if isinstance(this_attr, Node) and name not in do_not_copy:
-                    this_attr._setParent(self)
-            else:
-                copy_obj = deepcopy(getattr(_srcobj, name))
-                setattr(self, name, copy_obj)
+    def _setFlushed(self):
+        self._dirty = False
 
     def printTree(self, f=None, sel=''):
         from Ganga.GPIDev.Base.VPrinter import VPrinter
         self.accept(VPrinter(f, sel))
 
-    #printPrepTree is only ever run on applications, from within IPrepareApp.py
-    #if you (manually) try to run printPrepTree on anything other than an application, it will not work as expected
-    #see the relevant code in VPrinter to understand why
-    def printPrepTree(self, f=None, sel='preparable' ):
-        ## After fixing some bugs we are left with incompatible job hashes. This should be addressd before removing
```

Comment:
But now we're still left with ```printSummaryTree``` shouldn't the IPrepareApp overload this in some way rather than have its own summary tree method and we can just replace all ```printPrepTree``` back to ```printSummaryTree```?